core/src/main/java/hudson/model/PasswordParameterDefinition.java

```diff
@@ -66,6 +70,9 @@ public ParameterValue createValue(String value) {
     @Override
     public PasswordParameterValue createValue(StaplerRequest req, JSONObject jo) {
         PasswordParameterValue value = req.bindJSON(PasswordParameterValue.class, jo);
+        if (value.getValue().getPlainText().equals(DEFAULT_VALUE)) {
```

Comment:
I'm not 100% convinced value.getValue() never returns null.
BTW it needs some API usage investigation and proper annotations/Javadocs